lib/active_merchant/billing/gateways/xpay.rb

```diff
@@ -228,7 +223,7 @@ def commit(action, params, options)
       def request_headers(options, action = nil)
         headers = {
           'X-Api-Key' => @api_key,
-          'Correlation-Id' => options.dig(:order_id) || SecureRandom.uuid,
+          'Correlation-Id' => SecureRandom.uuid,
```

Comment:
⚠️ Warning / ❓ Question:

Any reason to remove `order_id` as the idempotency key ?, that value is not something that should be Random for each request, implementation should be along the lines of:

order_id => authorize, purchase, verify
authorization_value => refund, capture

maybe we should confirm with product on this one ...

Reply:
💬 Comment / Style:

Maybe a case for just one option is too much when we know that is unlikely to have more operations shortly:

```ruby
 headers.merge!('Idempotency-Key' => SecureRandom.uuid) if %i[refund capture].include?(action)
```

Reply:
Few days ago the Nexi Xpay SandBox works with the order_id in the place of the Correlation-Id, now we need to deal with separated fields, I will update the changes to support both